### DISCUSSION TITLE
How to generate weld path with collision-free tool orientation?

### DISCUSSION THREAD

**Discussion by savolla** [tesseract-robotics/tesseract#744] [Q&A] [answered]
Hi, I'm trying to build a 8 DOF gantry robot. I want to use Tesseract and Trajopt for path planning. I inspected and understood the code API by looking at [tesseract_ros_workcell](https://github.com/tesseract-robotics/tesseract_ros_workcell). But in the **application_positioner** example I saw that the robot is using pre-generated tool path trajectory points which is stored in [job_path.yaml](https://github.com/tesseract-robotics/tesseract_ros_workcell/blob/master/twc_application/config/job_path.yaml). I'd like to learn where this yaml file is generated. In addition since I want to build a welding robot instead of a sanding robot, I want to generate toolpath frames in collision-free manner. I saw some ros industrial repos like godel and bezier but It seems like they all generate a surface (planar) toolpath frames. Any explanation would be very helpful thanks!

**Comment by Levi-Armstrong** [comment, marked as answer]
Currently tesseract does not support toolpath generation directly and expects the path to be provided. There are a few opensource option available for tool path generation. 
- Generating toolpaths on surface and edges of surfaces you could use the [noether](https://github.com/ros-industrial/noether) package which was used to generate the toolpath in the job_path.yaml. 
  - There is a [plugin](https://github.com/swri-robotics/toolpath_offline_planning) for RViz which leverages noether
- There is also bezier but as you pointed out I believe it only does surfaces.
- There is also tools like [FreeCAD](https://www.freecadweb.org/index.php) which have a CNC workbech for generating toolpath from cad models.
- You could also leverage [OpenCascad](https://www.opencascade.com/) tools to generated toolpaths from cad.

**Reply by savolla** [reply]
Thank you! Currently working with noether as you said.
